src/Entity/EntityId.php

```diff
@@ -17,6 +18,30 @@
 
 	protected $serialization;
 
+	const PATTERN = '/^:?(\w+:)*[^:]+\z/';
```

Comment:
Wait, this regular expression **does** allow `:a:b:Q1`, but the expressions in the subclasses do not. I believe this should not be allowed, to make things as simple as possible and not have multiple string representations (`a:Q1` and `:a:Q1`) for the exact same ID.